bikeshed/InputSource.py

```diff
@@ -0,0 +1,155 @@
+from __future__ import annotations
+
+import email.utils
+import errno
+import io
+import os
+import sys
+import urllib.parse
+from abc import abstractmethod
+from dataclasses import dataclass
+from datetime import date, datetime
+from typing import List, Optional
+
+import requests
+
+from .Line import Line
+
+
+@dataclass
```

Comment:
Very similar to attr, but:

1. attr grew support for mypy in version 18.2.0, but bikeshed embeds (?!?) 17.3.0.
2. [dataclass doesn't support slots](https://www.python.org/dev/peps/pep-0557/#support-for-automatically-setting-slots). :(
3. PyRight, the VS Code default python formatter/typechecker, [doesn't intend to support attr](https://github.com/microsoft/pyright/issues/146).

Probably the right thing to do is to import a newer attr from PyPi, which I'll send you another PR to do.

Reply:
#1672 is merged, so feel free to swap this back over to attr

Reply:
Done.